ldes-server-port-admin-rest/src/main/java/be/vlaanderen/informatievlaanderen/ldes/server/admin/rest/controllers/AdminShapeRestController.java

```diff
@@ -39,7 +45,14 @@ public ResponseEntity<LdesConfigModel> getShape(@PathVariable String collectionN
 	public ResponseEntity<LdesConfigModel> putShape(@PathVariable String collectionName,
 			@RequestBody @Validated LdesConfigModel shape) {
 		LdesConfigModel updatedShape = service.updateShape(collectionName, shape);
+		eventPublisher.publishEvent(new ShaclChangedEvent(collectionName, shape.getModel()));
 		return ResponseEntity.ok(updatedShape);
 	}
 
+	@PostConstruct
+	private void initShapeConfig() {
+		service.retrieveAllShapes().stream()
+				.map(configModel -> new ShaclChangedEvent(configModel.getId(), configModel.getModel()))
+				.forEach(eventPublisher::publishEvent);
+	}
```

Comment:
Like the idea, have a suggestion for an alternative implementation. Maybe we could have a look later. Made an issue: https://github.com/Informatievlaanderen/VSDS-LDESServer4J/issues/527